tests/components/google_assistant/test_trait.py

```diff
@@ -387,6 +388,48 @@ async def test_locate_vacuum(hass):
     assert err.value.code == const.ERR_FUNCTION_NOT_SUPPORTED
 
 
+async def test_energystorage_vacuum(hass):
+    """Test EnergyStorage trait support for vacuum domain."""
+    assert helpers.get_google_type(vacuum.DOMAIN, None) is not None
+    assert trait.EnergyStorageTrait.supported(
+        vacuum.DOMAIN, vacuum.SUPPORT_BATTERY, None, None
+    )
+
+    trt = trait.EnergyStorageTrait(
+        hass,
+        State(
+            "vacuum.bla",
+            vacuum.STATE_DOCKED,
+            {
+                ATTR_SUPPORTED_FEATURES: vacuum.SUPPORT_BATTERY,
+                ATTR_BATTERY_LEVEL: 80,
```

Comment:
Also here: is it worth testing for multiple battery levels?

Reply:
No not really. Possibly edge case of full since.